Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerListModel.cpp

```diff
@@ -850,6 +850,12 @@ namespace AzToolsFramework
             return false;
         }
 
+        // Disable parenting to a different owning instance
+        if (m_focusModeInterface->IsInFocusSubTree(newParentId) != m_focusModeInterface->IsInFocusSubTree(selectedEntityIds.front()))
```

Comment:
I'm not sure if relying on outliner selection entity logic to stay the same is a good idea. If the outliner logic changes in the future to account for more features like overrides, anything will be possible. So, selecting whether all entities are in the focus tree will be a better error-proof solution imo. If it shows up as a bottleneck in profiling, we can try to make the logic smarter but fort starters, I think we should be checking all entities and not just the first one.

Reply:
I'm sure this code will go through more iterations once we get override support and make new decisions on what is allowed to be selected/reparented and when, but right now only focus mode is supported and the nature of focus mode is that you can only multiply select from the prefab that is in focus. If later we decide we want to support a drag & copy operations onto the focused prefab, I don't even think we would need or want a restriction of having all selected entities be under the same instance. So to summarize, IMO we shouldn't spend too much effort on bullet proofing scenarios that can't happen or trying to predict future behavior since the code is in the process of being iterated on anyway.

Having said that, I updated the PR to put the logic under a "focus mode" check since that is the mode where we want to ensure that only the focused prefab is modified, and I also checked all selected entities to satisfy review feedback (even though I don't believe it's necessary).